src/App.tsx

```diff
@@ -1,10 +1,46 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import { Todo } from './types/Todo';
+import { ErrorMessage } from './types/ErrorMessage';
+import { StatusFilter } from './types/StatusFilter';
+import { getTodos } from './api/todos';
+import { Header } from './components/Header/Header';
+import { TodoList } from './components/TodoList/TodoList';
+import { Footer } from './components/Footer/Footer';
+import { ErrorNotification } from './components/ErrorNotification';
 
-const USER_ID = 0;
+const USER_ID = 11968;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState<ErrorMessage | null>(null);
+  const [statusFilter, setStatusFilter] = useState(StatusFilter.ALL);
+
+  const activeTodos = todos.filter(todo => !todo.completed).length;
+  const completedTodos = todos.filter(todo => todo.completed).length;
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(setTodos)
+      .catch(() => {
+        setErrorMessage(ErrorMessage.LOAD_ERROR);
+      });
+  }, []);
+
+  const todosToRender = useMemo(() => todos.filter(todo => {
+    switch (statusFilter) {
+      case StatusFilter.ACTIVE:
+        return !todo.completed;
+
+      case StatusFilter.COMPLETED:
+        return todo.completed;
+
+      default:
+        return true;
+    }
+  }), [todos, statusFilter]);
```

Comment:
use filter inside switch case not vice versa to avoid extra iteration in default case